tests/run-pass/arithmetic_with_left_recursion.rs

```diff
@@ -0,0 +1,19 @@
+extern crate peg;
+
+use arithmetic::sum;
+
+peg::parser!( grammar arithmetic() for str {
+    #[cache]
+    pub rule sum() -> i64
+        = l:sum() "+" r:number() { l+r }
+        / number()
+
+    rule number() -> i64
+        = n:$(['0'..='9']+) { n.parse().unwrap() }
+});
+
+fn main() {
+    assert_eq!(sum("1+1"), Ok(2));
```

Comment:
`"1"` that falls through directly into `number` would be another good test case.

Reply:
Definitely.